scene/gui/color_picker.cpp

```diff
@@ -139,34 +138,66 @@ void ColorPicker::_notification(int p_what) {
+				sliders[i]->add_theme_icon_override("grabber", theme_cache.bar_arrow);
+				sliders[i]->add_theme_icon_override("grabber_highlight", theme_cache.bar_arrow);
+				sliders[i]->add_theme_constant_override("grabber_offset", slider_arrow_size.height);
+				sliders[i]->add_theme_style_override("slider", slider_style);
 				sliders[i]->add_theme_constant_override(SNAME("center_grabber"), theme_cache.center_slider_grabbers);
+				sliders[i]->end_bulk_theme_override();
+				labels[i]->set_custom_minimum_size(Size2(theme_cache.label_width, 0));
 			}
-			alpha_label->set_custom_minimum_size(Size2(theme_cache.label_width, 0));
+
+			alpha_slider->begin_bulk_theme_override();
+			alpha_slider->add_theme_icon_override("grabber", theme_cache.bar_arrow);
+			alpha_slider->add_theme_icon_override("grabber_highlight", theme_cache.bar_arrow);
+			alpha_slider->add_theme_constant_override("grabber_offset", slider_arrow_size.height);
+			alpha_slider->add_theme_style_override("slider", slider_style);
 			alpha_slider->add_theme_constant_override(SNAME("center_grabber"), theme_cache.center_slider_grabbers);
+			alpha_slider->end_bulk_theme_override();
+			alpha_label->set_custom_minimum_size(Size2(theme_cache.label_width, 0));
 			intensity_label->set_custom_minimum_size(Size2(theme_cache.label_width, 0));
 
-			for (int i = 0; i < MODE_BUTTON_COUNT; i++) {
+			for (int i = 0; i < MODE_MAX; i++) {
 				mode_btns[i]->begin_bulk_theme_override();
 				mode_btns[i]->add_theme_style_override(SceneStringName(pressed), theme_cache.mode_button_pressed);
 				mode_btns[i]->add_theme_style_override(CoreStringName(normal), theme_cache.mode_button_normal);
 				mode_btns[i]->add_theme_style_override(SceneStringName(hover), theme_cache.mode_button_hover);
 				mode_btns[i]->end_bulk_theme_override();
 			}
 
+			text_type->set_button_icon(text_is_constructor ? theme_cache.color_script : theme_cache.color_hex);
+
+			int margin = MAX(theme_cache.content_margin, 0);
 			internal_margin->begin_bulk_theme_override();
-			internal_margin->add_theme_constant_override(SNAME("margin_bottom"), theme_cache.content_margin);
-			internal_margin->add_theme_constant_override(SNAME("margin_left"), theme_cache.content_margin);
-			internal_margin->add_theme_constant_override(SNAME("margin_right"), theme_cache.content_margin);
-			internal_margin->add_theme_constant_override(SNAME("margin_top"), theme_cache.content_margin);
+			internal_margin->add_theme_constant_override(SNAME("margin_bottom"), margin);
+			internal_margin->add_theme_constant_override(SNAME("margin_left"), margin);
+			internal_margin->add_theme_constant_override(SNAME("margin_right"), margin);
+			internal_margin->add_theme_constant_override(SNAME("margin_top"), margin);
 			internal_margin->end_bulk_theme_override();
 
-			_reset_sliders_theme();
+			Size2 cursor_size = (theme_cache.picker_cursor->get_size() * 0.5).round();
+			real_vbox->add_theme_constant_override(SNAME("separation"), cursor_size.height);
```

Comment:
This doesn't work well.

<img width="181" height="414" alt="Image" src="https://github.com/user-attachments/assets/6d2c7057-7d4b-4509-becc-22819d31440c" />

It makes sense for the shape margins, but not for the whole ColorPicker. You could add a MarginContainer for this effect (if there isn't one that can be used already).